src/inputs/Button/Button.tsx

```diff
@@ -1,44 +1,52 @@
-import { PropsWithChildren } from "react";
-import * as React from "react";
-
+import React, { PropsWithChildren } from "react";
 import styles from "./Button.module.css";
+import { Spinner } from "../../icons";
+import { Fragment } from "react";
+import classNames from "classnames/bind";
+const cx = classNames.bind(styles);
 
-export interface ButtonProps {
-  onClick(): void;
+interface ButtonProps {
+  type?: "button" | "submit";
   disabled?: boolean;
-  variant?: "primary" | "secondary";
-  isLoading?: boolean;
+  isLoading: boolean;
+  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  variant: "primary" | "secondary";
+  icon: React.ReactNode;
+  children: React.ReactNode;
 }
 
 export const Button = ({
+  type = "button",
   disabled = false,
-  variant = "primary",
-  ...props
+  isLoading,
+  variant,
+  icon,
+  children,
 }: PropsWithChildren<ButtonProps>) => {
-  const variantClass =
-    variant === "primary" ? styles.primary : styles.secondary;
-
-  const getDisabledClassForSpecificVariant = (): string | void => {
-    if (disabled) {
-      return variant === "primary"
-        ? styles.disabledPrimary
-        : styles.disabledSecondary;
-    }
-  };
-
-  const dynamicClasses = [
-    styles.template,
-    variantClass,
-    getDisabledClassForSpecificVariant(),
-  ].join(" ");
+  const buttonClasses = cx({
+    buttons: true,
+    primary: variant === "primary",
+    secondary: variant === "secondary",
+    disabled: disabled,
+    isLoadingPrimary: variant === "primary" && isLoading && !disabled,
+    isLoadingSecondary: variant === "secondary" && isLoading,
+  });
 
   return (
-    <button
-      className={dynamicClasses}
-      onClick={props.onClick}
-      disabled={disabled}
-    >
-      {props.children}
+    <button className={buttonClasses} type={type} disabled={disabled}>
```

Comment:
**issue**: `isLoading` looks, and behaves different than on design. It should not have hover, and box shadow

<img width="653" alt="Zrzut ekranu 2023-04-7 o 18 54 31" src="https://user-images.githubusercontent.com/22893188/230647140-cd75fc09-3bf4-4557-830b-4587e0932c09.png">